nifi-nar-bundles/nifi-graph-bundle/nifi-graph-processors/src/main/java/org/apache/nifi/processors/graph/ExecuteGraphQueryRecord.java

```diff
@@ -0,0 +1,265 @@
+    public Set<Relationship> getRelationships() {
+        return RELATIONSHIPS;
+    }
+
+    @Override
+    public List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return DESCRIPTORS;
+    }
+
+    private GraphClientService clientService;
+    private RecordReaderFactory recordReaderFactory;
+    private RecordSetWriterFactory recordSetWriterFactory;
+    private ObjectMapper mapper = new ObjectMapper();
+
+    @OnScheduled
+    public void onScheduled(ProcessContext context) {
+        clientService = context.getProperty(CLIENT_SERVICE).asControllerService(GraphClientService.class);
+        recordReaderFactory = context.getProperty(READER_SERVICE).asControllerService(RecordReaderFactory.class);
+        recordSetWriterFactory = context.getProperty(WRITER_SERVICE).asControllerService(RecordSetWriterFactory.class);
+        recordPathCache = new RecordPathCache(context.getProperties().size() * 100);
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        FlowFile input = session.get();
+        if ( input == null ) {
+            return;
+        }
+
+        FlowFile output = session.create(input);
+        boolean failed = false;
+        final AtomicLong errors = new AtomicLong();
+        long delta = 0;
+        Map<String, String> attributes = new HashMap<>();
+        List<Map<String,Object>> graphResponses = new ArrayList<>();
+        try (InputStream is = session.read(input);
+             OutputStream os = session.write(output);
+             RecordReader reader = recordReaderFactory.createRecordReader(attributes, is, -1l, getLogger());
+             RecordSetWriter writer = recordSetWriterFactory.createWriter(getLogger(), reader.getSchema(), os, input.getAttributes());
+        ) {
+            Record record;
+
+            long start = System.currentTimeMillis();
+            long recordIndex = 0;
+            long batchStart = start;
+            String recordScript = getScript(SUBMISSION_SCRIPT, context, input);
+            writer.beginRecordSet();
+            while ((record = reader.nextRecord()) != null) {
+                ConcurrentMap<String, List<Object>> dynamicPropertyMap = new ConcurrentHashMap<>();
+                for (PropertyDescriptor entry : context.getProperties().keySet()) {
+                    if(entry.isDynamic()) {
+                        if(!dynamicPropertyMap.containsKey(entry.getName())) {
+                            String valueRecordPath = context.getProperty(entry.getName()).evaluateAttributeExpressions(input).getValue();
+                            final RecordPath recordPath = recordPathCache.getCompiled(valueRecordPath);
+                            final RecordPathResult result = recordPath.evaluate(record);
+                            List<Object> resultSet = result.getSelectedFields().filter(fv -> fv.getValue() != null).map(FieldValue::getValue).collect( Collectors.toList());
```

Comment:
This block here should be its own function call to reduce the complexity of the `onTrigger` method.

Reply:
Refactored with a functional call to this block of code.